core/src/main/scala/chisel3/choice/package.scala

```diff
@@ -47,4 +52,27 @@ package object choice {
       */
     def ->[T](module: => T): (Case, () => T) = (this, () => module)
   }
+
+  /** Registers all options in a group with the Builder.
+    * This lets Chisel know  that this layer should be emitted into FIRRTL text.
+    *
+    * This API can be used to guarantee that a design will always have certain
+    * group defined.  This is analagous in spirit to [[layer.addLayer]].
+    */
+  def addGroup[T <: Group: TypeTag](group: T): Unit = {
+
+    val tpe = typeOf[T]
+    val classSymbol = tpe.typeSymbol.asClass
+    val classMirror = cm.reflectClass(classSymbol)
+
+    tpe.members.collect {
+      // Look only for inner objects.  Note, this is not recursive.
+      case m: ModuleSymbol if m.isStatic =>
+        val instance = cm.reflectModule(m.asModule).instance
+        // Confirms the instance is a subtype of Case
+        if (cm.classSymbol(instance.getClass).toType <:< typeOf[Case]) {
+          Builder.options += instance.asInstanceOf[Case]
+        }
+    }
+  }
```

Comment:
This is clever reflection. 👍  However, it may be better to just make this information more easily accessible. Two changes would avoid the `TypeTag` (which is a Scala3 migration annoyance, I think) and any work in this function:

1. Add a `Builder` member for storing `groups` and not just options.
2. Add `mutable.LinkedHashSet` to `Group` that can be used to directly look up what `Case`s it supports. Then modify `Case` to insert into it.

Reply:
I started doing a (2) but wasn't sure when/how to finalize the set short of doing something either janky or switching the API in someway to use a different pattern. 

But (1) seems really easy to do; the existing struct can probably just become a HashMap and we wouldn't need to reconstruct the the groups later.

Reply:
Ok i got rid of the type tag but i can't figure out how to avoid it altogether unless we change the API for defining these cases (so that they are not lazy initialized).

Alternatively, we could change the `addGroup` API to to instead take a one or more cases. For example `addGroupHint(c: Case)` would suffice to workaround my problem, but it seems rather hacky to not register the whole group and breaks the direct `addLayer` analogy.

However, I doubt breaking the `Group API here is a big deal since no one should be using this? I'd change the `Group` implementation to use a more factory like pattern that would force registration on the definition of the objects.  

LMK

Reply:
Yeah, I think the laziness of the _cases_ seems wrong. Laziness of _groups_ seems fine. (You only get a group if you use it, but you get all of its cases.)

This would be a change to have the `Builder` register groups, have cases register that they are children of groups, and then pushing that change through to the converter.

However, all of this is independent of any change to the API. Given that, I will go ahead and approve this.

Reply:
I'm not sure if this reflection works in Scala 3 and as we are actively trying to get Scala 3 compilation into CI, I think we need to address it sooner rather than later. Can we do something similar to how `Layers` get the parents with an implicit rather than reflecting on the `Group`?

Reply:
Well, can i break the existing API? If so this is pretty straight forward. Otherwise i might need one of you guys (i.e., someone with more Scala foo than i) to do it. The crux of it is that the Cases are lazily constructed so _something_ has to reference them to bind them to the group. If you're coming into the office today maybe we can chat about it?

Reply:
You can break it as much as you want. Just don't backport it to 6.x. This feature has no users and you won't be disturbing anyone.